Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/utils/SettingsFile.java

```diff
@@ -23,6 +23,26 @@
 import java.util.TreeSet;
 
 /**
+ * A HashMap<String, SettingSection> that constructs a new SettingSection instead of returning null
+ * when getting a key not already in the map
+ */
+final class SettingsSectionMap  extends HashMap<String, SettingSection>
+{
+	@Override
+	public SettingSection get(Object key)
+	{
+		if (!(key instanceof String))
+		{
+			return null;
+		}
+		String stringKey = (String)key;
+		if (!super.containsKey(stringKey))
+			super.put(stringKey, new SettingSection(stringKey));
```

Comment:
Would it be better to return the instance being created here, instead of going through the map again (since the data is already known up front in this case)?

Reply:
Oh, I didn't know if the cost was significant or not, I was just offering it as a suggestion in case it was.

Reply:
I suspect "cost" isn't an issue here, it's in the ui querying a map that will have <10 entries that's not on any kind of hot path whatsoever. I guess it's more style than anything, I'm fine with either way.